src/github.com/getlantern/flashlight/ui/service.go

```diff
@@ -22,18 +22,17 @@ var (
 	mu               sync.RWMutex
 	defaultUIChannel *UIChannel
 
-	out      = make(chan []byte, 10)
 	services = make(map[string]*Service)
 )
 
-func (s *Service) watch() {
+func (s *Service) write() {
 	// Watch for new messages and send them to the combined output.
 	for msg := range s.out {
 		b, err := newEnvelope(s.Type, msg)
 		if err != nil {
 			log.Error(err.Error())
 		}
-		out <- b
+		defaultUIChannel.Out <- b
```

Comment:
This old out channel didn't go anywhere, so the UI wasn't getting updates and this would eventually block once the channel was filled.  We should just write directly to the defaultUIChannel.Out.